src/stores/ConstraintStore.ts

```diff
@@ -0,0 +1,58 @@
+import { makeAutoObservable, runInAction } from "mobx";
+
+export interface Constraint {
+  id: string
+  label: string
+  parameters: { [key: string]: string }[]
+};
+
+function renameKey<T extends Record<string, any>>(obj: T, oldKey: string, newKey: keyof T): T {
+  if (oldKey === newKey) {
+    return obj;
+  }
+  const { [oldKey]: value, ...rest } = obj;
+  return { [newKey]: value, ...rest } as T;
+}
+
+function renameKeysInList<T extends Record<string, any>>(list: T[], oldKey: string, newKey: keyof T): T[] {
+  return list.map(item => renameKey(item, oldKey, newKey));
+}
+
+export class ConstraintStore {
+
+  availableConstraints: Constraint[] = [];
+  isLoading: boolean = false;
+  error: string = "";
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  async fetchData(config_path: string) {
+    this.isLoading = true;
+    this.error = "";
+    //TODO: figure out how to get the config.json from github
+    // const rewrittenConfigPath = config_path.replace("https://github.com/", "https://raw.githubusercontent.com/")
+    //     .replace("/tree/", "/") + "/config.json";
+    // "https://github.com/sanctuuary/APE_UseCases/tree/master/MassSpectometry"
+    // "https://raw.githubusercontent.com/Workflomics/domain-annotations/main/MassSpectometry/config.json"
+    
+    const response = await fetch(`/ape/get_constraints?config_path=${config_path}`);
+    const result = await response.json();
+    runInAction(() => {
+      this.isLoading = false;
+      if (result.error !== undefined) {
+        this.error = result.error;
+      }
+      else {
+        let mappedResult = renameKeysInList<Constraint>(result, "constraintID", "id");
```

Comment:
Do you think the API call should return the fields as "ID" instead of "constraintsID"?

Reply:
That would be more consistent indeed. The reason for the explicit rename is that it's going to be used in the tree selection widget.

Reply:
I understand the need, this was more question regarding potential improvements to the REST API w.r.t. the best practices.

Reply:
Also in that way it would make more sense to just call it 'id'. The object returned is already a constraint, so why repeat that in the field name? It would only make sense if it is a different type referring to a constraint object. 😉

Reply:
Exactly, that's why I asked